modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java

```diff
@@ -414,6 +435,45 @@ protected ObjectDefinition testGetObjectDefinitionsPage_addObjectDefinition(
 		return _addObjectDefinition(objectDefinition);
 	}
 
+	@Override
+	protected void testGetObjectDefinitionsPageWithFilter(
+			String operator, EntityField.Type type)
+		throws Exception {
+
```

Comment:
@murilostodolni we should call the super method here, no?

Reply:
hey @gabrielwas 

The `testGetObjectDefinitionsPageWithFilter` method is auto generated by the rest build. In this test, the filter is performed by the existing EntityTypes in the `ObjectDefinitionEntityModel` (the String types are `name` and `objectFolderExternalReferenceCode`). With the self-generated method, the filter by name works normally. However, with the ObjectFolder's business rule, this test does not work, as it creates two definitions and [expects only one definition to be returned in the get with pagination](https://github.com/liferay-objects/liferay-portal/pull/2445/files/af8d2c34a42fdb358ed81287ef81fffb5867b719..32154fce2aabeebc04a626515b333d3b73aa119d#diff-c0571cda8d488d17dc5aafb818eaee8e045a1b2a4ce929d9aec5ef6b0095d364:~:text=assertEquals(,.getItems())%3B). In the case of objectFolder, if we use the self-generated method, all definitions that are in _"uncategorized"_ will always be returned, since they are added to _"uncategorized"_ by default. For that reason, we need to _override_ the method and add two different objectFolders for each definition created [here](https://github.com/liferay-objects/liferay-portal/pull/2445/files/af8d2c34a42fdb358ed81287ef81fffb5867b719..32154fce2aabeebc04a626515b333d3b73aa119d#diff-c0571cda8d488d17dc5aafb818eaee8e045a1b2a4ce929d9aec5ef6b0095d364:~:text=ObjectDefinition%20objectDefinition1%20%3D,objectDefinition1)%3B) and [here](https://github.com/liferay-objects/liferay-portal/pull/2445/files/af8d2c34a42fdb358ed81287ef81fffb5867b719..32154fce2aabeebc04a626515b333d3b73aa119d#diff-c0571cda8d488d17dc5aafb818eaee8e045a1b2a4ce929d9aec5ef6b0095d364:~:text=ObjectDefinition%20objectDefinition2%20%3D,(objectDefinition2)%3B). This way, when filtering by `ObjectFolderExternalReferenceCode`, only a definition will be returned, as expected in the test. Did you unterstand? Thanks